Plugin/test1/src/main/java/test/OkHttpInstrumentationAction.java

```diff
@@ -0,0 +1,191 @@
+package test;
+
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.command.WriteCommandAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.psi.*;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import util.InstrumentationResultMessage;
+import util.InstrumentationUtil;
+
+import java.util.List;
+
+public class OkHttpInstrumentationAction extends AnAction {
+    private static final int STATEMENT_TYPE_DECLARATION = 0;
+    private static final int STATEMENT_TYPE_ASSIGNMENT = 1;
+    private static final int STATEMENT_TYPE_RETURN = 2;
+
+    private Project project;
+    private InstrumentationResultMessage resultMessage;
+
+    /**
+     * Checks the existence of okHttp variables in this project AND Instruments to get OkHttp
+     *
+     * @param e {@inheritDoc}
+     */
+    @Override
+    public void actionPerformed(AnActionEvent e) {
+        project = e.getProject();
+        resultMessage = new InstrumentationResultMessage();
+        String[] fileFilter = new String[]{"import okhttp3"};
+        String[] classFilter = new String[]{"OkHttpClient"};
+
+        List<PsiFile> psiFiles = InstrumentationUtil.getAllJavaFilesInProjectAsPsi(project);
+
+        InstrumentationUtil.runScanOnJavaFile(psiFiles, fileFilter, classFilter, this::processPsiStatement);
+
+        Messages.showMessageDialog(resultMessage.getMessage(), "OkHttp Instrumentation Result ", Messages.getInformationIcon());
+    }
+
+    /**
+     * Scan a statement to search for a instance of OkHttpClient to instrument.
+     * This method is used as {@link java.util.function.Consumer} callback for the method
+     * {@link util.InstrumentationUtil#runScanOnJavaFile}
+     * <br/><br/>
+     *
+     * <p>The following occurrences should be instrumented </p>
+     *
+     * <pre>
+     * {@code
+     * OkHttpClient client = new OkHttpClient()
+     * OkHttpClient client = okHttpClientBuilder.build()
+     * client = new OkHttpClient()
+     * client = okHttpClientBuilder.build()
+     * }
+     * </pre>
+     *
+     * @param rootPsiElement A potential Java statement to instrument
+     */
+    private void processPsiStatement(@NotNull PsiElement rootPsiElement) {
+        rootPsiElement.accept(new JavaRecursiveElementVisitor() {
+            @Override
+            public void visitElement(PsiElement element) {
+                String test = element.getText();
```

Comment:
Remove this string before submitting the PR
It has been useful for debugging purposes, so I don't need to evaluate the expression all the time